### PR TITLE
feat: leverage isactive for transaction

### PR DESCRIPTION
**Pull request by jannyHou**
Implements https://github.com/strongloop/loopback-next/issues/3471
Previous PRs see:
https://github.com/strongloop/loopback-connector/pull/164
https://github.com/strongloop/loopback-connector/pull/165

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by jannyHou** [comment]
Checking why postgresql doesn't remove transaction's `connection` when it's released

**Comment by jannyHou** [comment]
@nabdelgadir I released a new version for `loopback-connector`, which contains a new function that this PR depends on, so I also updated the dependencies in the package-lock.json file to fetch the latest `loopback-connector`

**Comment by jannyHou** [comment]
The `tx` on [this line](https://github.com/strongloop/loopback-next/blob/master/packages/repository-tests/src/crud/transactions.suite.ts#L93) is an instance of different `Transaction` for postgresql and mysql tests:


The [`tx` created by postgresql connector](https://github.com/strongloop/loopback-connector-postgresql/blob/master/lib/transaction.js#L31) is an instance of `Transaction` in 

`packages/repository/node_modules/lb-connector/transaction.js`

The [`tx` created by mysql connector](https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L110) is an instance of `Transaction` in 

`packages/repository/node_modules/lb-datasource-juggler/transaction.js`, which should be the expected behavior

**Comment by jannyHou** [comment]
When transaction is created from [lb-connector's `Transaction.begin`](https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L110), the `Transaction`'s `__proto__` has juggler's transaction methods:

Screen Shot 2020-01-22 at 2 44 18 PM

But when transaction is created from [lb-connector-postgresql's `beginTransaction`](https://github.com/strongloop/loopback-connector-postgresql/blob/master/lib/transaction.js#L31), the `Transaction`'s `__proto__` DOES NOT include juggler's transaction methods:

Screen Shot 2020-01-22 at 2 40 47 PM

So the problem turns to be lb-connector's function [`Transaction`](https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L22) behaves differently when invoked by different callers.

**Comment by jannyHou** [comment]
Found the reason.

`packages/repository` has a juggler and its loopback-connector as dep, and juggler's Transaction mixin is applied to its lb-connector's Transaction. 
So tests in mysql [creates transaction](https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L110) with the Transaction that has mixin's overriden prototype methods.

While for postgresql, [`create transaction`](https://github.com/strongloop/loopback-connector-postgresql/blob/master/lib/transaction.js#L31) uses the Transaction in `acceptance/repository-postgresql`'s dependency, which doesn't have the juggler's Transaction mixin applied. And therefore won't [delete the connection reference for `tx`](https://github.com/strongloop/loopback-datasource-juggler/blob/251ced811b80177f47efb6a3270ed6d279263c46/lib/transaction.js#L151)

File: packages/repository/src/transaction.ts [hunk 1]
```diff
@@ -17,6 +17,11 @@ export interface Transaction {
    */
   rollback(): Promise<void>;
 
+  /**
+   * Check if the transaction has an active connection
+   */
+  isActive(): Promise<boolean>;
```
Comment (raymondfeng) [inline packages/repository/src/transaction.ts]: Is it `boolean` or `Promise<boolean>`?

File: packages/repository-tests/src/crud/transactions.suite.ts [hunk 2]
```diff
@@ -71,12 +71,16 @@ export function transactionSuite(
...
+          // For postgresql connector, it doesn't wrap the connection with
+          // the Transaction that has juggler's Transaction mixin applied,
+          // therefore `commit` or `rollback` won't delete the connection's
+          // reference of a `tx`.
+          // Detailed explanation see:
+          // https://github.com/strongloop/loopback-next/pull/4474
+          if (ds.connector && ds.connector.name === 'postgresql') {
+            tx = undefined;
+          }
+          if (tx?.isActive()) {
```
Comment (raymondfeng) [inline packages/repository-tests/src/crud/transactions.suite.ts]: This test will be always `true` if `tx.isActive()` returns a promise.

Comment (jannyHou) [reply packages/repository-tests/src/crud/transactions.suite.ts]: @raymondfeng good catch!
it is not a promise, fixed.

**Pull request merged**